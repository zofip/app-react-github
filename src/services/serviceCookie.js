import Cookies from 'universal-cookie';
const cookies = new Cookies();

export function saveCookie(name, value) {
  cookies.set(name, value, { path: '/' });
}

export function getCookie(name) {
  return cookies.get(name);
}