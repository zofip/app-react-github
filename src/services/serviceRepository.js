import { SLASH, REPOS } from "../util/constants";

export function getValueByUrl(url) {
  return fetch(url, {
    method: "GET"
  });
}

export function getRepositoriesByUser(user) {
  return fetch(process.env.REACT_APP_API_GET_REPOSITORIES + user + SLASH + REPOS, {
    method: "GET"
  });
}