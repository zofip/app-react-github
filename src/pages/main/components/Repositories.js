import React, { useEffect, useState } from "react";

import { getRepositoriesByUser, getValueByUrl } from "../../../services/serviceRepository";
import { getCookie } from "../../../services/serviceCookie";
import { INFO_USER, EMPTY } from "../../../util/constants";
import { Grid, TextField, makeStyles } from "@material-ui/core";
import MaterialTable from "material-table";

const useStyles = makeStyles(theme => ({
  textField: {
    marginLeft: theme.spacing(1),
    marginRight: theme.spacing(1),
    width: 350
  },
}));

const Repositories = props => {

  const classes = useStyles();
  const [fullData, setFullData] = useState([]);
  const [data, setData] = useState([]);
  const [infoUser] = useState(getCookie(INFO_USER));
  const [inputSearch, setInputSearch] = useState(EMPTY);
  const columns = [
    { title: "Name", field: "name" },
    { title: "Description", field: "description" },
    { title: "Languages", field: "languages_url" },
    { title: "Branch Default", field: "default_branch" },
    { title: "Git Url", field: "git_url" }
  ];

  useEffect(() => {
    if (infoUser) {
      getRepositoriesByUser(infoUser.user)
        .then(response => response.json())
        .then(repositories => {
          repositories.forEach(item => {
            getDetailRepository(item);
          });
          setData(repositories);
          setFullData(repositories);
        });
    }
  }, [infoUser]);

  const getDetailRepository = repository => {
    getValueByUrl(repository.languages_url)
      .then(response => response.json())
      .then(value => {
        repository.languages_url = Object.keys(value).toString();
      });
  }

  const handleChange = event => {
    let value = event.target.value;
    setInputSearch(value);
    if (value.length >= 3) {
      let filterRepositories = fullData.filter(item => item.name.includes(value));
      setData(filterRepositories);
    } else {
      setData(fullData);
    }
  };

  return (
    <div>
      <Grid container direction="row" spacing={5}>
        <Grid item xs={12}>
          <TextField
            id="inputSearch"
            label="Repository"
            value={inputSearch}
            className={classes.textField}
            onChange={handleChange} />
        </Grid>
        <Grid item xs={12} >
          {data.length > 0 && (
            <MaterialTable
              title='Repositories'
              columns={columns}
              data={data}
              options={{
                responsive: "scroll",
                search: false
              }}
            />
          )}
        </Grid>
      </Grid>
    </div>
  );
};

export default Repositories;
