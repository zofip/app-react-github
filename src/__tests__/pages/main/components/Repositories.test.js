import React from 'react';
import { shallow } from 'enzyme';
import { TextField } from '@material-ui/core';
import MaterialTable from 'material-table';

import Repositories from '../../../../pages/main/components/Repositories';
import { getRepositoriesByUser, getValueByUrl } from '../../../../services/serviceRepository';


describe('<Repositories />', () => {

    jest.fn(getRepositoriesByUser, () => {
        return [
            {
                name: "enzyme",
                description: "JavaScript Testing utilities for React",
                languages_url: "Javascript",
                default_branch: "master",
                git_url: "git://github.com/zofip/enzyme.git"
            },
        ]
    });

    jest.fn(getValueByUrl, () => {
        return [
            {
                java: "15",
                javascript: "1011"
            },
        ]
    });
    const repositories = shallow(<Repositories />);

    it('Rendering', () => {
        expect(repositories).toMatchSnapshot();
    });

    it('InputSearch', () => {
        repositories.find(TextField).simulate("change", { target: { value: "oth" } });
        repositories.find(TextField).simulate("change", { target: { value: "ot" } });
        repositories.find(MaterialTable);
    });


});