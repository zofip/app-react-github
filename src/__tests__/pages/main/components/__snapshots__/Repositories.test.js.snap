// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Repositories /> Rendering 1`] = `
<div>
  <WithStyles(ForwardRef(Grid))
    container={true}
    direction="row"
    spacing={5}
  >
    <WithStyles(ForwardRef(Grid))
      item={true}
      xs={12}
    >
      <WithStyles(ForwardRef(TextField))
        className="makeStyles-textField-1"
        id="inputSearch"
        label="Repository"
        onChange={[Function]}
        value=""
      />
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      item={true}
      xs={12}
    />
  </WithStyles(ForwardRef(Grid))>
</div>
`;
